lib/src/dbs/node.rs

```diff
@@ -49,18 +49,18 @@ impl From<&Timestamp> for KeyTimestamp {
 	}
 }
 
-impl Add<Duration> for Timestamp {
+impl Add<&Duration> for &Timestamp {
 	type Output = Timestamp;
-	fn add(self, rhs: Duration) -> Timestamp {
+	fn add(self, rhs: &Duration) -> Timestamp {
```

Comment:
Is there a reason the `Timestamp` value is not a `Duration`?

Reply:
The Timestamp struct was created to indicate point in time. It may not be epoch, as it may record logical clocks as well.